src/socket.c

```diff
@@ -32,16 +32,15 @@
 void socket_init(socket_ctx *ctx, char *path) {
 	printf("initialize socket\n");
 	ctx->socket.sun_family = AF_UNIX;
-	strcpy(ctx->socket.sun_path, path);
+	strncpy(ctx->socket.sun_path, path, strlen(path));
```

Comment:
I have managed to copy & paste the solution I have to admit I am not sure if the approach is actually much better.